### PR TITLE
add servlet example

### PR DESCRIPTION
**Pull request by liweinan**
According to the discussion here: https://github.com/resteasy/resteasy/discussions/3323#discussioncomment-5427801, add a minimal example.

### CODE REVIEW

**Comment by liweinan** [comment]
@jamezp Could you please help to review this example when you have time? Thanks!

File: servlet-example/pom.xml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    <profiles>
+        <profile>
+            <id>default</id>
+            <activation>
+                <activeByDefault>true</activeByDefault>
+            </activation>
+            <properties>
+                <maven.compiler.source>17</maven.compiler.source>
+                <maven.compiler.target>17</maven.compiler.target>
+            </properties>
```
Comment (jamezp) [inline servlet-example/pom.xml:48]: Why require Java 17?

File: servlet-example/src/main/java/io/weli/test/resteasy/example/FooResource.java
Comment (jamezp) [inline servlet-example/src/main/java/io/weli/test/resteasy/example/FooResource.java:1]: This should have a copyright header

File: servlet-example/src/main/java/io/weli/test/resteasy/example/MyApplication.java
Comment (jamezp) [inline servlet-example/src/main/java/io/weli/test/resteasy/example/MyApplication.java:1]: This should have a copyright header

**Review by jamezp — CHANGES_REQUESTED** [review]
Can we add a test to this as well? I think Jetty has some embedded test options. Just seems like it might be good for the example, but it's not critical. We can look at that later.

File: servlet-example/pom.xml (hunk 1 above)
Comment (liweinan) [reply servlet-example/pom.xml:48]: I forget to cleanup it :p

File: servlet-example/src/main/java/io/weli/test/resteasy/example/FooResource.java
Comment (liweinan) [reply servlet-example/src/main/java/io/weli/test/resteasy/example/FooResource.java:1]: okay!

File: servlet-example/src/main/java/io/weli/test/resteasy/example/MyApplication.java
Comment (liweinan) [reply servlet-example/src/main/java/io/weli/test/resteasy/example/MyApplication.java:1]: yes sir!

**Comment by liweinan** [comment]
@jamezp all changes are done!

**Pull request merged**